scipy/spatial/transform/rotation.py

```diff
@@ -616,3 +617,14 @@ def as_euler(self, seq, degrees=False):
             angles = np.rad2deg(angles)
 
         return angles[0] if self._single else angles
+
+    def inv(self):
+        """Returns the inverse of the current rotation
+
+        This function returns a new `Rotation` instance containing the inverse
+        of all the rotations in the current instance.
+        """
+        quat = self._quat.copy()
+        # (x,y,z,w) -> (x,y,z,-w)
+        quat[:, -1] *= -1
+        return self.__class__(quat, normalized=True)
```

Comment:
My concern that it discards `self._single` property. For now I suggest to fix it, however I also suggest to consider removing this flag all together and considering `(1, 4)` arrays as single automatically (ask this on the mail). The reasoning that this is uncommon case and it will simplify our code in several places.

Reply:
Can you elaborate on your suggestion a little?
- We could remove the `_single` flag completely, and forget about treating single rotations separately. The user could still initialize `from_quaternion` with a 1D array for example (all `from_*` functions will still behave in the same way) but the removal of this flag would mean that the `as_*` functions always return object with shape `(1,4)` or `(1,3,3)` or `(1, *..)`, regardless of how the user initialized it.
- Or do you mean to say that we check every time how many rotations are present in our object, and we appropriately reshape our return values from the `as_*` functions before returning? If this is the case, this is exactly what the flag achieves, for a minimal amount of extra memory, same condition check, and I think the same amount of code complexity. 

If I'm not mistaken, I think you mean the first option but I'd like to be sure before asking Eric and Ralf.

Reply:
I meant that if we have shape `(1, 4)` then we return a 1d array for `as_...` function. This way if a user initializes an object say using `from_quat([[0, 0, 0, 1]])` then it will behave somewhat inconsistently. However I don't think that such initialization will occur often (doesn't make too much sense), so we can be a little inconsistent, but simplify our code a little.